### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by elenapetra**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-  const [courseList, setCourseList] = useState<CourseData[]>(mockedCoursesList);
-
-  const updateCourses = (newCourse: CourseData) => {
-    setCourseList((courseList) => [...courseList, newCourse]);
-  };
-  const updateAuthors = (newAuthor: AuthorData) => {
-    setAuthorList((authorList) => [...authorList, newAuthor]);
-  };
+  useEffect(() => {
+    fetchAndAddCourses(dispatch);
```
Comment (dav-dio) [inline src/App.tsx]: nit: I don't like that you have to pass down `dispatch`, it looks like an anti-pattern.
What I'd advise instead is to create a custom hook that abstracts away this `useEffect` call. In a custom hook, you can call other hooks.
So it'd be something like:

```
export const useFetchAppData = async () => {
  const dispatch = useDispatch();
useEffect(() => {
  try {
    const authorsResponse = await fetchAuthors();
    dispatch(addAuthorAction(authorsResponse.result));
  } catch (error) {
    console.error('Error fetching and saving authors:', error);
  }

try {
    const coursesResponse = await fetchCourses();
    dispatch(addCourseAction(coursesResponse.result));
  } catch (error) {
    console.error('Error fetching and saving courses:', error);
  }
}, []);
};
```

and then you just call the hook in App.tsx.

File: src/common/Button/Button.tsx [hunk 2]
```diff
@@ -9,8 +9,9 @@ type ButtonProps = {
     | 'medium'
     | 'large'
     | 'super-large'
-    | 'extra-large';
-  label: string;
+    | 'extra-large'
+    | 'costum-btn';
```
Comment (dav-dio) [inline src/common/Button/Button.tsx]:
```suggestion
    | 'custom-btn';
```

Search and replace everywhere.

File: src/common/Button/Button.tsx
Comment (dav-dio) [inline src/common/Button/Button.tsx:1]: I really dislike how you're passing down explicit widths to buttons, especially numbers as random as 287 px.
Button widths should be determined mostly by their content. You can add extra padding if you feel like the button's edges are too close to the content, but there's no reason to specify the exact widths.

You don't need to fix this if you're really pressed on time (you should focus on Redux), but if you're going to be writing CSS or customizing UI components, please be aware that this is not good.

File: src/components/Courses/Components/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ export const SearchBar = ({ onSearch }: SearchBarProps) => {
         value={searchTerm}
         type='text'
         placeholder='Input text'
-        onChange={(e) => setSearchTerm(e.target.value)}
+        onChange={(e) => setSearchTerm(e.target.value.toLowerCase())}
```
Comment (dav-dio) [inline src/components/Courses/Components/SearchBar/SearchBar.tsx]: You shouldn't be converting the search term to lowercase _inside of the search bar_, you should just be comparing the lowercase version of the search term with the lowercase version of the course title or ID. It feels unnatural (bad UX) to not be able to type capital letters in a search box.

File: src/components/CreateCourse/CreateCourse.tsx
Comment (dav-dio) [inline src/components/CreateCourse/CreateCourse.tsx:1]: Please validate against adding a course without any authors.

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
+import { useSelector, useDispatch } from 'react-redux';
 import Button from 'common/Button/Button';
 import { Logo } from './components/Logo/Logo';
+import { removeUserAction } from 'store/user/actions';
+import { getUserName } from 'store/selectors';
 import './Header.css';
-import { useNavigate, useLocation } from 'react-router-dom';
 
 export const Header = () => {
   const [isLogedIn, setIsLogedIn] = useState(false);
```
Comment (dav-dio) [inline src/components/Header/Header.tsx]:
```suggestion
  const [isLoggedIn, setIsLoggedIn] = useState(false);
```

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
   const location = useLocation();
 
+  const handleLogout = () => {
+    localStorage.removeItem('token');
+    dispatch(removeUserAction());
+    navigate('/login');
+  };
+
   useEffect(() => {
     setIsLogedIn(!!token);
```
Comment (dav-dio) [inline src/components/Header/Header.tsx]: nit: You might be able to remove this useEffect and use `!!token` as the initial value of the `isLoggedIn` state:

```
const token = localStorage.getItem('token');
const [isLogedIn, setIsLogedIn] = useState(!!token);
```

Haven't tested it though

File: src/store/user/actions.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { UsersActionTypes, UserType, AddUserAction } from './types';
+
+export const addUserAction = (userData: UserType): AddUserAction => ({
```
Comment (dav-dio) [inline src/store/user/actions.ts]: nit: I find the naming to misleading. This is used for login, not registration, so you're not "adding" the user to any database (aside from the store). Same goes for courses and authors, to a lesser degree - you're calling "add courses" during the initial App.tsx load. I'd go with "setUser" or "storeUser," etc.

**Review by dav-dio — APPROVED** [review]

File: src/store/user/actions.ts (hunk 6 above)
Comment (elenapetra) [reply src/store/user/actions.ts]: I changed the naming for courses, authors and user : from add... to store...

File: src/components/Header/Header.tsx (hunk 5 above)
Comment (elenapetra) [reply src/components/Header/Header.tsx]: I removed the useState and useEffect.

File: src/components/Header/Header.tsx (hunk 4 above)
Comment (elenapetra) [reply src/components/Header/Header.tsx]: changed

File: src/components/CreateCourse/CreateCourse.tsx
Comment (elenapetra) [reply src/components/CreateCourse/CreateCourse.tsx:1]: I added validation

File: src/components/Courses/Components/SearchBar/SearchBar.tsx (hunk 3 above)
Comment (elenapetra) [reply src/components/Courses/Components/SearchBar/SearchBar.tsx]: Fixed

File: src/common/Button/Button.tsx (hunk 2 above)
Comment (elenapetra) [reply src/common/Button/Button.tsx]: changed

File: src/App.tsx (hunk 1 above)
Comment (elenapetra) [reply src/App.tsx]: I have created the custom hook and called it in the App.tsx

**Pull request merged**